backend/corpora/common/entities/dataset_asset.py

```diff
@@ -72,6 +79,8 @@ def create(
         user_submitted: bool,
         s3_uri: str,
     ):
+        from . import Dataset
```

Comment:
This will also be a problem when you do delete collections

Reply:
because the s3 deletion function is on the asset you cant call it without creating the asset. 

I'm not sure I understand why this will be an issue when deleting a collection